entity-framework/core/what-is-new/ef-core-8.0/whatsnew.md

```diff
@@ -15,768 +15,1285 @@ EF8 is available as [daily builds](https://github.com/dotnet/efcore/blob/main/do
 
-        public int Id { get; private set; }
+A good way to deal with issues like this it to make the type immutable. Indeed, this is often natural when a type is a good candidate for being a complex type. For example, it usually makes sense to supply a complex new `Address` object rather than to just mutate, say, the country while leaving the rest the same.
 
-        [Column("Title")]
-        public string BlogTitle { get; set; }
+Both reference and value types can be made immutable. We'll look at some examples in the following sections.
 
-        public string Content { get; set; }
-        public DateOnly PublishedOn { get; set; }
-        public int BlogId { get; set; }
-    }
--->
-[!code-csharp[BlogPost](../../../../samples/core/Miscellaneous/NewInEFCore8/RawSqlSample.cs?name=BlogPost)]
+### Reference types as complex types
 
-> [!NOTE]
-> Types used in this way do not have keys defined and cannot have relationships to other types. Types with relationships must be mapped in the model.
+#### Immutable class
 
-The type used must have a property for every value in the result set, but do not need to match any table in the database. For example, the following type represents only a subset of information for each post, and includes the blog name, which comes from the `Blogs` table:
+We used a simple, mutable `class` in the example above. To prevent the issues with accidental mutation described above, we can make the class immutable. For example:
 
-<!--
-    public class PostSummary
+```csharp
+public class Address
+{
+    public Address(string line1, string? line2, string city, string country, string postCode)
     {
-        public string BlogName { get; set; }
-        public string PostTitle { get; set; }
-        public DateOnly PublishedOn { get; set; }
+        Line1 = line1;
+        Line2 = line2;
+        City = city;
+        Country = country;
+        PostCode = postCode;
     }
--->
-[!code-csharp[PostSummary](../../../../samples/core/Miscellaneous/NewInEFCore8/RawSqlSample.cs?name=PostSummary)]
 
-And can be queried using `SqlQuery` in the same way as before:
+    public string Line1 { get; }
+    public string? Line2 { get; }
+    public string City { get; }
+    public string Country { get; }
+    public string PostCode { get; }
+}
+```
 
-<!--
-        var summaries =
-            await context.Database.SqlQuery<PostSummary>(
-                    @$"SELECT b.Name AS BlogName, p.Title AS PostTitle, p.PublishedOn
-                    FROM Posts AS p
-                    INNER JOIN Blogs AS b ON p.BlogId = b.Id")
-                .ToListAsync();
--->
-[!code-csharp[SqlQueryJoin](../../../../samples/core/Miscellaneous/NewInEFCore8/RawSqlSample.cs?name=SqlQueryJoin)]
+> [!TIP]
+> With C# 12 or above, this class definition can be simplified using a primary constructor:
+> public class Address(string line1, string? line2, string city, string country, string postCo
```

Comment:
IMHO, we should just show the most modern syntax. Most customers think of .NET releases as one big whole.